multires-lyrics-search/README.md

```diff
@@ -1,109 +1,176 @@
 
-**NOTE**: This is what is used in the Docker image and is **required** if you want to build it (the Docker image) yourself. 
+### Full dataset 
+Begin by installing the Kaggle library if you haven't already done so.  You will also need to setup your API keys as [explained here](https://github.com/Kaggle/kaggle-api#api-credentials) .
+```sh
+pip install kaggle
+```
 
-If you want to use the full dataset, you can download it from kaggle (https://www.kaggle.com/neisse/scrapped-lyrics-from-6-genres).
-To get it, once you have your Kaggle Token in your system as described in (https://www.kaggle.com/docs/api), run:
+Running the following bash script should perform all the steps needed to download the full dataset. 
 
 ```bash
 bash get_data.sh
 ```
 
-## Install
+## Step 3. Index your data
 
-```bash
-pip install -r requirements.txt
+In this step, we will index our data. When we run the following command, we call our index Flow and pass our data through it. The Flow begins by breaking each song into sentences and then encoding those sentences using a language model. The data is then stored in an Index. 
+
+```sh
+python app.py index
 ```
+If you see the following output, it means your data has been correctly indexed.
+```sh
+Flow@5162[S]:flow is closed and all resources are released, current build level is 0
+```
+## Step 4. Query your indexed data
 
-## Run
+Next, we will deploy our query Flow. This Flow will accept a search input, break it down into sentences and encode it using the same language model as above. It then performs a nearest neighbor search and finds sentences in the index which are most similar to your query sentence. 
 
-| Command | Description |
-| :--- | :--- |
-| ``python app.py index`` | To index files/data |
-| ``python app.py search`` | To run query on the index |
-| ``python app.py dryrun`` | Sanity check on the topology |
+Run the following command and leave it open and running in your terminal. 
 
-## View in Browser
+```sh
+python app.py search
+```
+Open a new terminal window and run the following command to launch the web interface. 
 
 ```bash
 cd static
 python -m http.server
 ```
+Now you can open `http://0.0.0.0:8000/` in your browser and you should see a web interface. See the next section to understand your results. 
 
-Open `http://0.0.0.0:8000/` in your browser.
+# Using the search interface and understanding your results
+Let's begin by starting with a simple one-sentence query. For example, if you add the sentence 'I am very happy today; you should see a similar result. Each of these boxes you see on the right-hand side is a song in your dataset. Each highlighted sentence is a 'match.'  A match is a similar sentence, determined by how close two vectors are in embedding space.  If you don't understand the concept of embedding spaces, we suggest you check out this guide [here](https://towardsdatascience.com/neural-network-embeddings-explained-4d028e6f0526) and then return to this example after. 
```

Comment:
Maybe put a link to lyrics search youtube video as well so readers can see it in action